### PR TITLE
fix: using include in public speaker query

### PR DESCRIPTION
**Pull request by kushthedude**
Refers #3650

- Using include to fetch sessions correponding to speakers.
- Using FilterOptions to define filter rather than explicitly defining them in query object

### CODE REVIEW

**Comment by iamareebjamal** [comment]
Don't understand how it fixes anything?

**Comment by kushthedude** [comment]
> If there are 200 speakers and in aggregate all of them have 300 sessions, then 300 tracks and 300 microlocations, then 900 requests will be sent to server just to show the speaker names and picture. This is where include should be used instead of sending 900 requests to the server, but is still not needed.

**Comment by iamareebjamal** [comment]
Summarize the changes in the PR

**Comment by kushthedude** [comment]
> Summarize the changes in the PR

Done

**Comment by iamareebjamal** [comment]
> Using FilterOptions to define filter rather than explicitly defining them in query object

Yeah, so this confused me. Why?

**Comment by kushthedude** [comment]
For ease of reference and understanding, should I revert this change of let
it be?

File: app/routes/public/speakers.js [hunk 1]
```diff
@@ -3,34 +3,36 @@ import Route from '@ember/routing/route';
 export default Route.extend({
   async model() {
     const eventDetails = this.modelFor('public');
+    let filterOptions = [
```
Comment (iamareebjamal) [inline app/routes/public/speakers.js]: Don't `let` it be. It should be const

**Comment by iamareebjamal** [comment]
This increases load time of queries from 700 ms to 2 seconds. This is not the solution when all speakers are loaded. More than 200 speakers can be loaded in 1 go. 

Use pagination on speakers page and only load sessions when speaker detail is clicked on. And there include track and microlocation with session. Not sure where microlocation is being used now

**Comment by kushthedude** [comment]
> This increases load time of queries from 700 ms to 2 seconds. This is not the solution when all speakers are loaded. More than 200 speakers can be loaded in 1 go.
> 
> Use pagination on speakers page and only load sessions when speaker detail is clicked on. And there include track and microlocation with session. Not sure where microlocation is being used now

When the UI Accordion is expanded, Session is shown with it's microlocation.

**Comment by iamareebjamal** [comment]
Can't see microlocation in any single session in FOSSASIA Summit 2020. Please give an example

https://eventyay.com/e/a74665c3/speakers

**Comment by kushthedude** [comment]
> some time ago, we discussed the idea of lazy loading speakers too -- if an event like FOSSASIA summit iself has a 100 speakers, loading all 100 images in one go does not make sense as a user either can't see them until he scrolls.

@CosmicCoder96 should infinityLoader be implemented ?

**Comment by kushthedude** [comment]
This needs further review and changes.

**Pull request closed without merging**